delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/ScopedEventLog.scala

```diff
@@ -261,29 +261,38 @@ object ScopedEventLog {
             EntityDependencyStore.delete(ref, state.id) >> EntityDependencyStore.save(ref, state.id, dependencies)
           }
 
-        def persist(event: E, original: Option[S], newState: S): IO[Rejection, Unit] =
-          saveTag(event, newState)
-            .flatMap { tagQuery =>
-              val queries = for {
-                _ <- TombstoneStore.save(entityType, original, newState)
-                _ <- eventStore.save(event)
-                _ <- stateStore.save(newState)
-                _ <- tagQuery
-                _ <- deleteTag(event, newState)
-                _ <- updateDependencies(newState)
-              } yield ()
-              queries
-                .attemptSomeSqlState { case sqlstate.class23.UNIQUE_VIOLATION =>
-                  onUniqueViolation(id, command)
-                }
-                .transact(xas.write)
-            }
-            .hideErrors
+        def persist(event: E, original: Option[S], newState: S): IO[Rejection, Unit] = {
+
+          def queries(tagQuery: ConnectionIO[Unit], cache: Set[String]) =
+            for {
+              _ <- TombstoneStore.save(entityType, original, newState)
+              _ <- eventStore.save(event, cache)
+              _ <- stateStore.save(newState, cache)
+              _ <- tagQuery
+              _ <- deleteTag(event, newState)
+              _ <- updateDependencies(newState)
+            } yield ()
+
+          {
+            for {
+              cache    <- xas.cache.get
+              tagQuery <- saveTag(event, newState)
```

Comment:
I think it would be better to generate the hashes only once here
We need to introduce a `PartitionAction` with a `Create` and a `Noop`, create holding the two hashes